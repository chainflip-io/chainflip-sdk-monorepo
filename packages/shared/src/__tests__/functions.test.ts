import {
  calculateHundredthPipAmountFromAmount,
  calculatePipAmountFromAmount,
} from '../functions';

describe(calculatePipAmountFromAmount, () => {
  it.each([
    ['10000000', 100, 100000n],
    ['250', 100, 2n],
    ['0', 100, 0n],
    ['1', 100, 0n],
  ])('calculates pip amount from amount', (amount, pips, result) => {
    expect(calculatePipAmountFromAmount(amount, pips)).toBe(result);
  });
});

describe(calculateHundredthPipAmountFromAmount, () => {
  it.each([
    ['10000000', 100, 1000n],
    ['10000000', 1000, 10000n],
    ['250', 100, 0n],
    ['0', 100, 0n],
    ['1', 100, 0n],
  ])(
    'calculates hundredth pip amount from amount',
    (amount, hundredthPips, result) => {
      expect(calculateHundredthPipAmountFromAmount(amount, hundredthPips)).toBe(
        result,
      );
    },
  );
});
