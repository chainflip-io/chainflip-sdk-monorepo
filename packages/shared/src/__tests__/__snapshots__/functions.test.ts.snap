// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseFoKParams > accepts the retry duration in 0 minutes 1`] = `
{
  "minPriceX128": "83892489958826316385497263710123985244108278726772",
  "refundAddress": "0x1234",
  "retryDurationBlocks": 0,
}
`;

exports[`parseFoKParams > accepts the retry duration in 0.5 minutes 1`] = `
{
  "minPriceX128": "83892489958826316385497263710123985244108278726772",
  "refundAddress": "0x1234",
  "retryDurationBlocks": 5,
}
`;

exports[`parseFoKParams > accepts the retry duration in 0.25 minutes 1`] = `
{
  "minPriceX128": "83892489958826316385497263710123985244108278726772",
  "refundAddress": "0x1234",
  "retryDurationBlocks": 3,
}
`;

exports[`parseFoKParams > accepts the retry duration in 1 minutes 1`] = `
{
  "minPriceX128": "83892489958826316385497263710123985244108278726772",
  "refundAddress": "0x1234",
  "retryDurationBlocks": 10,
}
`;

exports[`parseFoKParams > accepts the retry duration in 10 minutes 1`] = `
{
  "minPriceX128": "83892489958826316385497263710123985244108278726772",
  "refundAddress": "0x1234",
  "retryDurationBlocks": 100,
}
`;
